lang/java/reef-webserver/src/main/java/org/apache/reef/webserver/HttpServerImpl.java

```diff
@@ -76,17 +82,18 @@
    * @throws Exception
    */
   @Inject
-  HttpServerImpl(final JettyHandler jettyHandler,
-                 final LocalAddressProvider addressProvider,
+  HttpServerImpl(@Parameter(RemoteConfiguration.HostAddress.class) final String hostAddress,
                  @Parameter(TcpPortRangeBegin.class)final int portNumber,
+                 final JettyHandler jettyHandler,
+                 final LocalAddressProvider addressProvider,
                  final TcpPortProvider tcpPortProvider,
                  final LoggingScopeFactory loggingScopeFactory) throws Exception {
-    this.addressProvider = addressProvider;
     this.loggingScopeFactory = loggingScopeFactory;
     this.jettyHandler = jettyHandler;
     int availablePort = portNumber;
     Server srv = null;
 
+    this.hostAddress = UNKNOWN_HOST_NAME.equals(hostAddress) ? addressProvider.getLocalAddress() : hostAddress;
```

Comment:
This should be taken care of by the addressProvider.

Reply:
None of the LocalAddressProvider implementations abstract this. This is similar to how it is in NettyMessagingTransport.

---
In reply to: [199970265](https://github.com/sharathmalladi/reef/pull/6#discussion_r199970265) [](ancestors = 199970265)